algorithms/gr-bfsrh/csharp/BreadthFirst.cs

```diff
@@ -0,0 +1,96 @@
+using System.Collections.Generic;
+
+namespace Algorithms
+{
+    //Based on https://www.geeksforgeeks.org/shortest-distance-two-cells-matrix-grid/
+    //Binary-first search
+    public class BreadthFirst
+    {
+        private const int BLOCKED = 1;
+        private const int START = 2;
+        private const int END = 3;
+
+        public static int MinDistance(int[][] grid)
```

Comment:
Codacy found an issue: [Make this method private or simplify its parameters to not use multidimensional arrays.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6319078)